### PR TITLE
Add Sync feature flags

### PR DESCRIPTION
**Pull request by ayoy**
Task/Issue URL: https://app.asana.com/0/0/1206046777189407/f

**Description**:
This change add support for Sync feature in Privacy Configuration, together with
4 subfeatures defining availability of various parts of Sync experience.
DDGSyncing gets a read-only feature flag variable as well as a publisher.
PrivacyConfigurationManager is now a Sync dependency, and DDGSync takes
care internally of listening to Privacy Config changes and updating feature flags
as needed.
Feature flag responsible for actual data syncing is handled internally in DDGSync
by cancelling all pending sync operations and disabling adding new operations.
Other feature flags should be handled by client apps.

**Steps to test this PR**:
1. Fetch existing privacy config files for iOS and macOS.
2. Serve them from a different location.
3. Run each app, update Privacy Config URL (for macOS this can be done in runtime via Debug Menu).
4. Manipulate feature flags and verify that they work according to the specification.

**Copy Testing**:

* [ ] Use of correct apostrophes in new copy, ie `’` rather than `'`

**Orientation Testing**:

* [ ] Portrait
* [ ] Landscape

**Device Testing**:

* [ ] iPhone SE (1st Gen)
* [ ] iPhone 8
* [ ] iPhone X
* [ ] iPhone 14 Pro
* [ ] iPad

**OS Testing**:

* [ ] iOS 14
* [ ] iOS 15
* [ ] iOS 16

**Theme Testing**:

* [ ] Light theme
* [ ] Dark theme

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

File: Core/FeatureFlag.swift [hunk 1]
```diff
@@ -39,8 +39,10 @@ public enum FeatureFlag: String {
 extension FeatureFlag: FeatureFlagSourceProviding {
     public var source: FeatureFlagSource {
         switch self {
-        case .debugMenu, .sync, .appTrackingProtection:
+        case .debugMenu, .appTrackingProtection:
             return .internalOnly
+        case .sync:
+            return .remoteReleasable(.subfeature(SyncSubfeature.level0ShowSync))
```
Comment (bwaresiak) [inline Core/FeatureFlag.swift:45]: Shouldn't this be `.feature(.sync)` ?

Comment (ayoy) [reply Core/FeatureFlag.swift:45]: I'm not sure eventually. We decided that the top level flag will never be adjusted (and that it's only for the rollout), and that the level0 flag has the same meaning as the top-level feature. I'm totally fine with updating it to the top-level.

Comment (bwaresiak) [reply Core/FeatureFlag.swift:45]: Yea, I think it is fine, as this subfeature is also guarded by the main feature state. Let's keep it. :)

**Review by bwaresiak — APPROVED** [review]
Great job!

**Pull request merged**